[WPE] Remove OpenXR based WebXR implementation

svillar opened the pull request:
#### 9a901cc020174a6ba593cde3c4803d8476281295
<pre>
[WPE] Remove OpenXR based WebXR implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=294618">https://bugs.webkit.org/show_bug.cgi?id=294618</a>

Reviewed by Adrian Perez de Castro.

WebXR implementation using OpenXR for the WPE port has been
unmaintained for years. The current code does not work and
has architectural and design issues that recommend us to
remove it from the tree. As an example, it requires breaking
the WebProcess sandbox as OpenXR requires filesystem access
(to load the OpenXR loader).

We&apos;ll proceed to re-upload it progressively but mimicking the
design of the Cocoa implementation, i.e, we&apos;ll move the
OpenXR code to the UIProcess.

* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/xr/openxr/OpenXRExtensions.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRExtensions.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInput.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInput.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInputMappings.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInstance.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRLayer.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRLayer.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRSwapchain.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRSwapchain.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRUtils.h: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h: Removed.
* Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/296365@main">https://commits.webkit.org/296365@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd2cd96b651cbb957a286c1cfbe9724179ec5551

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108194 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27857 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18277 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113405 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58679 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110157 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28557 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36408 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82164 "Found 2 new test failures: fast/dom/navigator-detached-no-crash.html imported/w3c/web-platform-tests/feature-policy/reporting/xr-reporting.https.html (failure)") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111142 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22628 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97465 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62595 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22043 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15601 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58142 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/100783 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91994 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15657 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116531 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/106755 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35260 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25966 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91177 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35634 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93741 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90972 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23205 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35863 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13627 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/31040 "Build was cancelled. Recent messages:Printed configuration") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35159 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40715 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/131043 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34893 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35571 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38251 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36560 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/bd2cd96b651cbb957a286c1cfbe9724179ec5551. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46859

aperezdc commented:
FWIW, the EWS failures seem unrelated to this patch.

svillar committed:
[WPE] Remove OpenXR based WebXR implementation
https://bugs.webkit.org/show_bug.cgi?id=294618

Reviewed by Adrian Perez de Castro.

WebXR implementation using OpenXR for the WPE port has been
unmaintained for years. The current code does not work and
has architectural and design issues that recommend us to
remove it from the tree. As an example, it requires breaking
the WebProcess sandbox as OpenXR requires filesystem access
(to load the OpenXR loader).

We'll proceed to re-upload it progressively but mimicking the
design of the Cocoa implementation, i.e, we'll move the
OpenXR code to the UIProcess.

* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/xr/openxr/OpenXRExtensions.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRExtensions.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInput.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInput.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInputMappings.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInstance.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRLayer.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRLayer.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRSwapchain.cpp: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRSwapchain.h: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRUtils.h: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h: Removed.
* Source/cmake/OptionsWPE.cmake:

Canonical link: https://commits.webkit.org/296365@main
